packages/tools/devtools/devtools-core/src/data-visualization/SharedTreeVisualizer.ts

```diff
@@ -78,18 +99,44 @@ function createToolTipContents(schema: SharedTreeSchemaNode): VisualTreeNode {
 }
 
 /**
- * Constructs a VisualTree of the input tree's schema fields in {@link VisualTreeNode} or {@link VisualValueNode}.
+ * Converts the visual representation from {@link visualizeSharedTreeNodeBySchema} to a visual tree compatible with the devtools-view.
+ * @param tree - the visual representation of the SharedTree.
+ * @returns - the visual representation of type {@link VisualChildNode}
  */
-export function toVisualTree(tree: VisualSharedTreeNode): VisualValueNode | VisualTreeNode {
+export function toVisualTree(tree: VisualSharedTreeNode): VisualChildNode {
 	if (tree.kind === VisualSharedTreeNodeKind.LeafNode) {
-		const result: VisualValueNode = {
-			value: tree.value,
-			nodeKind: VisualNodeKind.ValueNode,
-			tooltipContents: {
-				schema: createToolTipContents(tree.schema),
-			},
-		};
-		return result;
+		switch (tree.value.nodeKind) {
+			case VisualNodeKind.ValueNode: {
+				const result: VisualValueNode = {
+					value: tree.value.value,
+					nodeKind: VisualNodeKind.ValueNode,
+					tooltipContents: {
+						schema: createToolTipContents(tree.schema),
+					},
+				};
+				return result;
+			}
+			case VisualNodeKind.FluidHandleNode: {
+				const result: FluidHandleNode = {
+					fluidObjectId: tree.value.fluidObjectId,
+					nodeKind: VisualNodeKind.FluidHandleNode,
+					tooltipContents: {
+						schema: createToolTipContents(tree.schema),
+					},
+				};
+				return result;
+			}
+			default: {
+				console.error(`Unknown node kind: ${tree.value.nodeKind}`);
```

Comment:
Nit: since this error would be resulting from logic we control (the creation of the visual tree), I think it would be appropriate to throw an error here.